File: src/Modal/Prog5/Qfactor/Qfactor.java
package Modal.Prog5.Qfactor;

import java.applet.Applet;
import java.awt.BorderLayout;
import java.awt.Frame;
import java.awt.Image;
import java.awt.Label;
import java.awt.TextField;
import java.util.Date;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import javax.swing.Timer;

@SuppressWarnings("serial")
public class Qfactor extends Applet implements ActionListener {
   public Frame mFeedbackWindow = null;
   public TextField answer1;
   static final double kMink = 150000.0;
   static final double kMaxk = 450000.0;
   static final double kMinm = 0.5;
   static final double kMaxm = 2.0;
   static final double kMinc = 1.0;
   static final double kMaxc = 500.0;
   CFramePanel mFramePanel;
   CFrameGraph mFrameGraph;
   CFrameAnimation mAnimFrame;
   Timer mTimer = null;
   long mLastTime = 0;

   public void init() {
      this.setLayout(new BorderLayout());
      Label copyright = new Label("  This program is copyright, you need permission to use it. © 2005 B J Stone, bjs@mech.uwa.edu.au.");
      this.add("South", copyright);
      this.mFramePanel = new CFramePanel(this);
      this.add("Center", this.mFramePanel);
      this.mFrameGraph = new CFrameGraph(this.mFramePanel, 0, 20, 1641, 365);
      CFrameSmallControl mControl = new CFrameSmallControl(this.mFramePanel, this.mFrameGraph, 2, 480, 40, 0.5, 1.0, 2.0, "kg");
      mControl.SetLabelOffset(20, 10);
      mControl.SetTextEditable(true);
      Image mImage = this.getImage(this.getCodeBase(), "m.gif");
      new CFramePicture(this.mFramePanel, 505, 55, 0, 0, mImage, false);
      CFrameSmallControl cControl = new CFrameSmallControl(this.mFramePanel, this.mFrameGraph, 3, 570, 40, 1.0, 50.0, 500.0, "Ns/m");
      cControl.SetLabelOffset(20, 10);
      cControl.SetTextEditable(true);
      Image cImage = this.getImage(this.getCodeBase(), "c.gif");
      new CFramePicture(this.mFramePanel, 595, 55, 0, 0, cImage, false);
      CFrameSmallControl kControl = new CFrameSmallControl(this.mFramePanel, this.mFrameGraph, 1, 480, 90, 150000.0, 400000.0, 450000.0, "N/m");
      kControl.SetLabelOffset(20, 10);
      kControl.SetTextEditable(true);
      Image kImage = this.getImage(this.getCodeBase(), "k.gif");
      new CFramePicture(this.mFramePanel, 505, 105, 0, 0, kImage, false);
      Image XaxisImage = this.getImage(this.getCodeBase(), "Xaxis.gif");
      new CFramePicture(this.mFramePanel, 129, 378, 0, 0, XaxisImage, false);
      Image YaxisImage = this.getImage(this.getCodeBase(), "Yaxis.gif");
      new CFramePicture(this.mFramePanel, 13, 13, 0, 0, YaxisImage, false);
      Image DiagramImage = this.getImage(this.getCodeBase(), "Diagram.gif");
      new CFramePicture(this.mFramePanel, 470, 135, 0, 0, DiagramImage, false);
      this.mAnimFrame = new CFrameAnimation(this.mFramePanel, this.mFrameGraph, 0, 385, this.getSize().width, 60);
      this.validate();
   }

   public void start() {
      this.mTimer = new Timer(50, this);
      this.mTimer.start();
   }

   public void actionPerformed(ActionEvent e) {
      if (this.mTimer != null) {
         Date ddd = new Date();
         long thisTime = ddd.getTime();
         
         if (this.mLastTime == 0) {
            this.mLastTime = thisTime;
         }
         double realSeconds = (thisTime - this.mLastTime) / 1000.0;
         this.mLastTime = thisTime;
         
         //this.mAnimFrame.ControlMessage(0, realSeconds);
      }
   }



   public void stop() {
      if (this.mTimer != null) { this.mTimer.stop(); this.mTimer = null; }
   }
}

File: site/swingjs/j2s/Modal/Prog5/Qfactor/Qfactor.js
(function(){var P$=Clazz.newPackage("Modal.Prog5.Qfactor"),I$=[[0,'java.awt.BorderLayout','java.awt.Label','Modal.Prog5.Qfactor.CFramePanel','Modal.Prog5.Qfactor.CFrameGraph','Modal.Prog5.Qfactor.CFrameSmallControl','Modal.Prog5.Qfactor.CFramePicture','Modal.Prog5.Qfactor.CFrameAnimation','javax.swing.Timer']],I$0=I$[0],$I$=function(i,n){return((i=(I$[i]||(I$[i]=Clazz.load(I$0[i])))),!n&&i.$load$&&Clazz.load(i,2),i)};
/*c*/var C$=Clazz.newClass(P$, "Qfactor", null, 'java.applet.Applet', 'java.awt.event.ActionListener');

C$.$clinit$=2;

Clazz.newMeth(C$, '$init$', function () {
this.mFeedbackWindow=null;
this.mTimer=null;
this.mLastTime=0;
},1);

C$.$fields$=[['J',['mLastTime'],'O',['mFeedbackWindow','java.awt.Frame','answer1','java.awt.TextField','mFramePanel','Modal.Prog5.Qfactor.CFramePanel','mFrameGraph','Modal.Prog5.Qfactor.CFrameGraph','mAnimFrame','Modal.Prog5.Qfactor.CFrameAnimation','mTimer','javax.swing.Timer']]]

Clazz.newMeth(C$, ['init$','init'],  function () {
this.setLayout$java_awt_LayoutManager(Clazz.new_($I$(1,1)));
var copyright=Clazz.new_($I$(2,1).c$$S,["  This program is copyright, you need permission to use it. \u00a9 2005 B J Stone, bjs@mech.uwa.edu.au."]);
this.add$S$java_awt_Component("South", copyright);
this.mFramePanel=Clazz.new_($I$(3,1).c$$java_applet_Applet,[this]);
this.add$S$java_awt_Component("Center", this.mFramePanel);
this.mFrameGraph=Clazz.new_($I$(4,1).c$$Modal_Prog5_Qfactor_CFramePanel$I$I$I$I,[this.mFramePanel, 0, 20, 1641, 365]);
var mControl=Clazz.new_($I$(5,1).c$$Modal_Prog5_Qfactor_CFramePanel$Modal_Prog5_Qfactor_CFrame$I$I$I$D$D$D$S,[this.mFramePanel, this.mFrameGraph, 2, 480, 40, 0.5, 1.0, 2.0, "kg"]);
mControl.SetLabelOffset$I$I(20, 10);
mControl.SetTextEditable$Z(true);
var mImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "m.gif");
Clazz.new_($I$(6,1).c$$Modal_Prog5_Qfactor_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 505, 55, 0, 0, mImage, false]);
var cControl=Clazz.new_($I$(5,1).c$$Modal_Prog5_Qfactor_CFramePanel$Modal_Prog5_Qfactor_CFrame$I$I$I$D$D$D$S,[this.mFramePanel, this.mFrameGraph, 3, 570, 40, 1.0, 50.0, 500.0, "Ns/m"]);
cControl.SetLabelOffset$I$I(20, 10);
cControl.SetTextEditable$Z(true);
var cImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "c.gif");
Clazz.new_($I$(6,1).c$$Modal_Prog5_Qfactor_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 595, 55, 0, 0, cImage, false]);
var kControl=Clazz.new_($I$(5,1).c$$Modal_Prog5_Qfactor_CFramePanel$Modal_Prog5_Qfactor_CFrame$I$I$I$D$D$D$S,[this.mFramePanel, this.mFrameGraph, 1, 480, 90, 150000.0, 400000.0, 450000.0, "N/m"]);
kControl.SetLabelOffset$I$I(20, 10);
kControl.SetTextEditable$Z(true);
var kImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "k.gif");
Clazz.new_($I$(6,1).c$$Modal_Prog5_Qfactor_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 505, 105, 0, 0, kImage, false]);
var XaxisImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "Xaxis.gif");
Clazz.new_($I$(6,1).c$$Modal_Prog5_Qfactor_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 129, 378, 0, 0, XaxisImage, false]);
var YaxisImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "Yaxis.gif");
Clazz.new_($I$(6,1).c$$Modal_Prog5_Qfactor_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 13, 13, 0, 0, YaxisImage, false]);
var DiagramImage=this.getImage$java_net_URL$S(this.getCodeBase$(), "Diagram.gif");
Clazz.new_($I$(6,1).c$$Modal_Prog5_Qfactor_CFramePanel$I$I$I$I$java_awt_Image$Z,[this.mFramePanel, 470, 135, 0, 0, DiagramImage, false]);
this.mAnimFrame=Clazz.new_([this.mFramePanel, this.mFrameGraph, 0, 385, this.getSize$().width, 60],$I$(7,1).c$$Modal_Prog5_Qfactor_CFramePanel$Modal_Prog5_Qfactor_CFrameGraph$I$I$I$I);
this.validate$();
});

Clazz.newMeth(C$, ['start$','start'],  function () {
this.mTimer=Clazz.new_($I$(8,1).c$$I$java_awt_event_ActionListener,[50, this]);
this.mTimer.start$();
});

Clazz.newMeth(C$, ['actionPerformed$java_awt_event_ActionEvent','actionPerformed'],  function (e) {
if (this.mTimer != null ) {
var ddd=Clazz.new_(java.util.Date);
var thisTime=ddd.getTime$();
if (Long.$eq(this.mLastTime,0 )) {
this.mLastTime=thisTime;
}this.mLastTime=thisTime;
}});

Clazz.newMeth(C$, ['stop$','stop'],  function () {
if (this.mTimer != null ) {
this.mTimer.stop$();
this.mTimer=null;
}});

Clazz.newMeth(C$);
})();
;Clazz.setTVer('5.0.1-v7');//Created 2025-10-14 19:43:50 Java2ScriptVisitor version 5.0.1-v7 net.sf.j2s.core.jar version 5.0.1-v7
